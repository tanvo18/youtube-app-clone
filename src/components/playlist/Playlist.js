import React, { Component } from 'react';
import PlaylistStyled from './PlaylistStyled';
import { Link } from 'react-router-dom';
import Indicator from '../indicator/Indicator';
import PropTypes from 'prop-types';

class Playlist extends Component {

  componentDidMount() {
    this.props.requestPlaylist();
  }

  render() {
    const { playlist, isFetching } = this.props;

    return (
      <PlaylistStyled>
        <ul className="playlist">
          {
            isFetching
              ? <Indicator />
              : <div>
                {
                  playlist && playlist.length > 0
                    ? playlist.map((playlistVideo, index) => (
                      <li
                        key={index}
                        className="playlist-item"
                      >
                        <Link
                          className="video-link"
                          to={`/playlist-detail/${playlistVideo.id}`}
                        >
                          <div className="thumbnail-wrap">
                            {
                              playlistVideo.snippet.thumbnails
                                ? <img src={playlistVideo.snippet.thumbnails.medium.url} alt="" className="thumbnail-img" />
                                : <img src="" alt="" className="thumbnail-img" />
                            }

                          </div>
                          <div className="playlist-description-wrap">
                            <p className="title-description">{playlistVideo.snippet.title}</p>
                            <p className="quantity-description">VIEW FULL PLAYLIST ({playlistVideo.contentDetails.itemCount})</p>
                          </div>
                        </Link>
                      </li>
                    ))
                    : <p>There are no playlist</p>
                }
              </div>
          }
        </ul>
      </PlaylistStyled>
    );
  }
}

Playlist.defaultProps = {
  playlist: [],
};

Playlist.propTypes = {
  playlist: PropTypes.array,
};

export default Playlist;