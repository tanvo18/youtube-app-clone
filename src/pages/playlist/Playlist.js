import React, { Component } from 'react';
import PlaylistContainer from '../../containers/playlist/PlaylistContainer';

class Playlist extends Component {
  render() {
    return (
      <div>
        <PlaylistContainer />
      </div>
    );
  }
}

export default Playlist;