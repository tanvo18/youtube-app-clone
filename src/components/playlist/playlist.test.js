import React from 'react';
import Playlist from './Playlist';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

const props = {
  playlist: [
    {
      'id': 'PLYD4txFCoRIddpm1qaxj39HW0odA4Ojie',
      'snippet': {
        'publishedAt': '2018-02-17T22:36:32.000Z',
        'channelId': 'UCM6UtOuKhYDX2Q76rz_-Zug',
        'title': 'Soundtrack',
        'description': '',
        'thumbnails': {
          'medium': {
            'url': 'https://i.ytimg.com/vi/CuzRarStqAI/mqdefault.jpg',
            'width': 320,
            'height': 180
          },
        },
        'channelTitle': 'Tan Vo',
      },
      'contentDetails': {
        'itemCount': 35
      }
    },
  ]
};

const propsEmptyThumbnail = {
  playlist: [
    {
      'id': 'PLYD4txFCoRIddpm1qaxj39HW0odA4Ojie',
      'snippet': {
        'publishedAt': '2018-02-17T22:36:32.000Z',
        'channelId': 'UCM6UtOuKhYDX2Q76rz_-Zug',
        'title': 'Soundtrack',
        'description': '',
        'thumbnails': undefined,
        'channelTitle': 'Tan Vo',
      },
      'contentDetails': {
        'itemCount': 35
      }
    },
  ]
};

describe('test playlist', () => {
  it('test requestPlaylist function', () => {
    const requestPlaylist = jest.fn();
    const wrapper = shallow(<Playlist 
      requestPlaylist={requestPlaylist}
      {...props}
    />);
    wrapper.instance().componentDidMount();
    expect(requestPlaylist).toHaveBeenCalled();
  });

  it('test data is fetching', () => {
    const requestPlaylist = jest.fn();
    const isFetching = true;
    const wrapper = mount(<Playlist 
      requestPlaylist={requestPlaylist}
      isFetching={isFetching}
      {...props}
    />);
    expect(wrapper.find('Indicator').find('div').exists()).toBe(true);
    expect(wrapper.find('.playlist-item').exists()).toBe(false);
  });

  it('test data fetched successfully', () => {
    const requestPlaylist = jest.fn();
    const isFetching = false;
    const wrapper = mount(<Router><Playlist 
      requestPlaylist={requestPlaylist}
      isFetching={isFetching}
      {...props}
    /></Router>);
    expect(wrapper.find('Indicator').find('div').exists()).toBe(false);
    expect(wrapper.find('.playlist-item').exists()).toBe(true);
  });

  it('test empty data', () => {
    const props = {playlist: []};
    const requestPlaylist = jest.fn();
    const wrapper = shallow(<Playlist 
      requestPlaylist={requestPlaylist}
      {...props}
    />);
    expect(wrapper.find('p').text()).toBe('There are no playlist');
    expect(wrapper.find('.playlist-item').exists()).toBe(false);
  });

  it('empty thumbnail', () => {
    const requestPlaylist = jest.fn();
    const wrapper = shallow(<Playlist 
      requestPlaylist={requestPlaylist}
      {...propsEmptyThumbnail}
    />);

    expect(wrapper.find('.thumbnail-img').exists()).toBe(true);
  });

  it('should render correctly', () => {
    const requestPlaylist = jest.fn();
    const isFetching = false;
    const tree = renderer
      .create(<Router><Playlist 
        requestPlaylist={requestPlaylist}
        isFetching={isFetching}
        {...props}
      /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

