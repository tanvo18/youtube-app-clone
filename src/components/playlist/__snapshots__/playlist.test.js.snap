// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test playlist should render correctly 1`] = `
<div
  className="sc-bdVaJa ijSSsU"
>
  <ul
    className="playlist"
  >
    <div>
      <li
        className="playlist-item"
      >
        <a
          className="video-link"
          href="/playlist-detail/PLYD4txFCoRIddpm1qaxj39HW0odA4Ojie"
          onClick={[Function]}
        >
          <div
            className="thumbnail-wrap"
          >
            <img
              alt=""
              className="thumbnail-img"
              src="https://i.ytimg.com/vi/CuzRarStqAI/mqdefault.jpg"
            />
          </div>
          <div
            className="playlist-description-wrap"
          >
            <p
              className="title-description"
            >
              Soundtrack
            </p>
            <p
              className="quantity-description"
            >
              VIEW FULL PLAYLIST (
              35
              )
            </p>
          </div>
        </a>
      </li>
    </div>
  </ul>
</div>
`;
